### ISSUE TITLE
0.8.4 improvment request

### ISSUE THREAD

**Issue by pallebone** [ubergeek77/Lemmy-Easy-Deploy#56]
Hi There, it is me again.

Currently for lemmy easy deploy the tags are hard coded for:
Caddy,
Pictrs,
Postgres

This means we cant select our own tag if we need to for various reasons. I have personally experienced this limitation once in the past.
Can it be implemented that selecting a tag can be customizable if required?

Kind regards
Peter

**Comment by ubergeek77** [comment]
I'm not sure what is being asked of here. Do you mean the prefix that shows on the log output for the stack?

This is just how Docker Compose works. If you want to filter just the logs for a given service, you can do that, and you can also show those logs without any prefix at all:

https://stackoverflow.com/questions/50360674/how-can-i-remove-prefix-with-service-name-from-logs/68855292#68855292

Do you want a custom prefix because you want to clearly see the logs for different services? Because if so, you can just show them by service name as shown in that discussion.

**Comment by pallebone** [comment]
Interesting, this does make it difficult as certain features come out for each of the components quite rapidly. In addition if you use the latest tag for caddy (which you do) its not clear how you are ensuring everyone uses the identical version of caddy since depending when they deploy it a situation can occur where if caddy upload a new version, a different version is being deployed of it for 1 person using lemmy easy deploy to another.

**Comment by pallebone** [comment]
What did you think about this?

**Comment by ubergeek77** [comment]
I think that level of meticulously tracking every version of every service is out of scope for this project, as this is not professional software. You are still free to specify custom tags if you like.

**Comment by pallebone** [comment]
Alright can you tell me how I specify a custom docker compose template that is not overwritten when I update with the script? Im not clear how to complete this.

**Comment by ubergeek77** [comment]
Anything in the `custom` folder will never be overwritten on updates. You can check the documentation in the repo and copy the official template into that directory with the correct name, then edit it how you like.

**Comment by pallebone** [comment]
Ok so this seems doable so if I just make a copy of docker-compase.yml and put the copy exactly here: ./custom/docker-compose.yml.template

Then it will just use this from then on? This doesnt seem too bad as an option. I can try that if this is the solution? I assume this still conforms to your specification if I do it that way?

P

**Comment by ubergeek77** [comment]
That's correct. It will always refer to that file instead of the one I supply.

It "conforms" to my specification, but if you have any issues as a result of this deployment, I won't really be able to help as the deployment will now be non-standard.

**Comment by pallebone** [comment]
Alrighty it seems okay. May I also ask how the cloudflare template works as I would like to implement cloudflare for ddos. I dindnt realise I had to make lemmy easy deploy aware of cloudflare. What exactly does the cloudflare template do or tell lemmy to do?

**Comment by pallebone** [comment]
Ok thank you. Your expertise in this is very much valued. It makes sense what you are saying. I am going to need to do this for cloudflare so I may have to ask for some assistance next week when I set it up to check its working as it makes sense that if I enable the proxy for ddos it would interfere with letsencrypt getting the cert renewal. 

Regards stability yes this is my main motivation so of your advice here is that its ok then will still stick with what you have specified unless Im am forced to change for some reason (hopefully this wont happen). So far I think I am only using 2 custom files - one for extra pictrs variables for object storage and one for the postgres performance improvements so Im pretty sure I still conform 100%. This is why I have a lot of questions surrounding changes instead of just doing whatever without first asking. 

Thanks again. I noticed I am also using the debian images you create now as you built for amd64 so it seems I am pretty much as stable as possible through your script. 

Appreciate you taking the time to talk to me. 
P
